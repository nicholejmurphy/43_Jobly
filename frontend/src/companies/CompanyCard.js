import React from "react";
import { Link } from "react-router-dom";

/** Shows details about an individual company
 *  Links to Company details page
 *  - Route: "/companies/:handle"
 */
function CompanyCard({ handle, name, numEmployees, description, logoUrl }) {
  return (
    <div className="CompanyCard bg-light rounded shadow p-3 m-3">
      <Link to={`/companies/${handle}`}>
        <h2>{name}</h2>
      </Link>
      <hr className="w-25 ml-0" />
      <p className="lead">{description}</p>
      <p>
        Number of employees:{" "}
        <span className="badge badge-info">{numEmployees}</span>
      </p>
    </div>
  );
}

export default CompanyCard;
