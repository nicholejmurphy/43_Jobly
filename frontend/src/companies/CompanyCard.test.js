import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

// smoke test
it("renders without crashing", function () {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <CompanyCard />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
