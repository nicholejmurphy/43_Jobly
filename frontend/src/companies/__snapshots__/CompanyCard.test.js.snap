// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard bg-light rounded shadow p-3 m-3"
  >
    <a
      href="/companies/undefined"
    >
      <h2 />
    </a>
    <hr
      class="w-25 ml-0"
    />
    <p
      class="lead"
    />
    <p>
      Number of employees: 
      <span
        class="badge badge-info"
      />
    </p>
  </div>
</DocumentFragment>
`;
